app/src/main/java/com/duckduckgo/app/browser/camera/CameraHardwareChecker.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.camera
+
+import android.content.Context
+import android.content.pm.PackageManager.FEATURE_CAMERA_ANY
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface CameraHardwareChecker {
+    fun hasCameraHardware(): Boolean
```

Comment:
1. there is a hardware capability check in site-permissions code. https://github.com/duckduckgo/android/blob/dbbf1670142d84c04039112693f851a203a75d8a/site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsManagerImpl.kt#L54

1. Most, if not fully 100%, of the Android devices which run our app should have a camera, but it's possible some obscure device won't. It is Android after all, so best to handle that possibility just in case.